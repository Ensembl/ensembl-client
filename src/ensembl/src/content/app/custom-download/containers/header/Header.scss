@import 'src/styles/common';

.wrapper {
  background-color: $ens-light-grey;
  height: 60px;
  position: relative;
}

.resultCounter {
  position: absolute;
  top: 50%;
  left: 50px;
  color: $ens-black;
  margin-top: -10px;
}

.selectedFilters {
  position: absolute;
  top: 50%;
  left: 40%;
  margin-top: -12px;
}

.previewButton {
  position: absolute;
  right: 30px;
  top: 50%;
  margin-top: -15px;
}

/* RoundButton extended styles*/
.active {
  background-color: $ens-medium-dark-grey;
  border: 1px solid $ens-medium-dark-grey;
  color: $ens-white;
  margin-right: 10px;
}

.previewResultCounter {
  margin-left: 20px;
  padding-top: 12px;
  display: inline-block;
}

.boldResultCounter {
  font-weight: bold;
  margin-right: 2px;
}

.resultsLabel {
  font-size: 11px;
  color: $ens-dark-grey;
}
.saveConfiguration {
  font-size: 11px;
  color: $ens-blue;
}

.backButton {
  height: 30px;
  width: 30px;
  margin-left: 20px;
  display: inline-block;
}

.downloadTypeSelectHolder {
  display: inline-block;
  margin-left: 20px;
}

.downloadTypeLabel {
  color: $ens-dark-grey;
}

.downloadTypeSelect {
  margin-left: 10px;
  color: $ens-blue;
  li {
    color: $ens-medium-dark-grey;
  }
}
