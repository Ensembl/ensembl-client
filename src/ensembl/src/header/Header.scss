@import 'src/styles/common';

.topBar {
  @include top-bar-container;
  @include top-bar-unstack;

  color: $white;
  padding: 2px 20px;
}

.homeLink {
  display: inline-block;
  font-size: 14px;
  font-weight: $global-weight-bold;

  a {
    color: $ens-white;
  }

  img {
    bottom: 2px;
    height: 16px;
    margin-right: 2px;
    position: relative;
    width: 16px;
  }
}

.strapline,
.copyright {
  display: none;
  font-size: 11px;
  margin-left: 30px;

  @include breakpoint(medium) {
    display: inline-block;
  }

  a {
    color: $ens-blue;
  }
}
