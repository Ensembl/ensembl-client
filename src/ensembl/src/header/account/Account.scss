@import 'src/styles/common';

.account {
  border-bottom: 1px solid $ens-medium-grey;
  height: 300px;
  @include flex;
  @include flex-align(center, middle);

  h2 {
    color: $ens-grey;
    text-align: center;
  }
}
