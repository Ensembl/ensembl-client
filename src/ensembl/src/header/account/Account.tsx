import React, { FunctionComponent, ReactNode } from 'react';
import SlideDown from 'react-slidedown';
import { connect } from 'react-redux';

import { RootState } from 'src/rootReducer';
import { getAccountExpanded } from '../headerSelectors';

import styles from './Account.scss';

type StateProps = {
  accountExpanded: boolean;
};

type OwnProps = {};

type AccountProps = StateProps & OwnProps;

export const Account: FunctionComponent<AccountProps> = (
  props: AccountProps
) => {
  const AccountChildren: ReactNode = (
    <div className={styles.account}>
      <h2>Account area placeholder</h2>
    </div>
  );

  return (
    <SlideDown>{props.accountExpanded ? AccountChildren : null}</SlideDown>
  );
};

const mapStateToProps = (state: RootState): StateProps => ({
  accountExpanded: getAccountExpanded(state)
});

export default connect(mapStateToProps)(Account);
