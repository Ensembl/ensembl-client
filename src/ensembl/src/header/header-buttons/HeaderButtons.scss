.button {
  display: inline-block;
  bottom: 1px;
  height: 17px;
  margin-left: 20px;
  position: relative;
  width: 17px;
  vertical-align: middle;
}

.headerButtonDisabled {
  svg {
    fill: #7d8e9b;
  }
}
