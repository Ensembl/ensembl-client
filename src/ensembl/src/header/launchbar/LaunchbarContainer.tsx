import React, { FunctionComponent, memo } from 'react';
import { connect } from 'react-redux';

import { RootState } from 'src/store';
import Launchbar from './Launchbar';
import { getLaunchbarExpanded } from '../headerSelectors';

type StateProps = {
  launchbarExpanded: boolean;
};

type OwnProps = {};

type LaunchbarContainerProps = StateProps & OwnProps;

export const LaunchbarContainer: FunctionComponent<
  LaunchbarContainerProps
> = memo((props) => <Launchbar {...props} />);

const mapStateToProps = (state: RootState): StateProps => ({
  launchbarExpanded: getLaunchbarExpanded(state)
});

export default connect(mapStateToProps)(LaunchbarContainer);
