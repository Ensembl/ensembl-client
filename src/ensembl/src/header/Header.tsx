import React, { memo, FunctionComponent } from 'react';

import HeaderButtons from './header-buttons/HeaderButtons';
import LaunchbarContainer from './launchbar/LaunchbarContainer';
import Account from './account/Account';

import styles from './Header.scss';

type HeaderProps = {};

export const Header: FunctionComponent<HeaderProps> = () => (
  <header>
    <div className={styles.topBar}>
      <div>
        <div className={styles.companyText}>Ensembl</div>
        <div className={styles.strapline}>Pre-release - March 2019</div>
        <div className={styles.copyright}>
          <a href="https://www.ebi.ac.uk">&copy; EMBL-EBI</a>
        </div>
      </div>
      <HeaderButtons />
    </div>
    <Account />
    <LaunchbarContainer />
  </header>
);

export default memo(Header);
