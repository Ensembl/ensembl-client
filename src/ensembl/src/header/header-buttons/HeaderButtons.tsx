import React, { FunctionComponent } from 'react';
import { connect } from 'react-redux';

import { toggleAccount, toggleLaunchbar } from '../headerActions';

import ImageButton, {
  ImageButtonStatus
} from 'src/shared/image-button/ImageButton';

import { ReactComponent as LaunchbarIcon } from 'static/img/header/launchbar.svg';
import { ReactComponent as UserIcon } from 'static/img/header/user-grey.svg';

import styles from './HeaderButtons.scss';

type StateProps = {};

type DispatchProps = {
  toggleLaunchbar: () => void;
  toggleAccount: () => void;
};

type OwnProps = {};

type HeaderButtonsProps = StateProps & DispatchProps & OwnProps;

export const HeaderButtons: FunctionComponent<HeaderButtonsProps> = (props) => (
  <div className={styles.headerButtons}>
    <div className={styles.buttonWrapper}>
      <ImageButton
        image={LaunchbarIcon}
        description="Ensembl app launchbar"
        onClick={props.toggleLaunchbar}
      />
    </div>
    <div className={styles.buttonWrapper}>
      <ImageButton
        image={UserIcon}
        description="Ensembl account"
        buttonStatus={ImageButtonStatus.DISABLED}
        classNames={{
          [ImageButtonStatus.DISABLED]: styles.headerButtonDisabled
        }}
      />
    </div>
  </div>
);

const mapStateToProps = (): StateProps => ({});

const mapDispatchToProps: DispatchProps = {
  toggleAccount,
  toggleLaunchbar
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(HeaderButtons);
