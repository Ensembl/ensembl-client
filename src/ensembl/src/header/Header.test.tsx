import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Header from './Header';
import Account from './account/Account';
import LaunchbarContainer from './launchbar/LaunchbarContainer';
import HeaderButtons from './header-buttons/HeaderButtons';

describe('<Header />', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<Header />);
  });

  describe('displays', () => {
    test('Ensembl', () => {
      expect(wrapper.find('.companyText').text()).toBe('Ensembl');
    });

    test('strapline', () => {
      expect(wrapper.find('.strapline').text()).toBe(
        'Pre-release - March 2019'
      );
    });
  });

  describe('contains', () => {
    test('Account', () => {
      expect(wrapper.contains(<Account />)).toBe(true);
    });

    test('Launchbar', () => {
      expect(wrapper.contains(<LaunchbarContainer />)).toBe(true);
    });

    test('HeaderButtons', () => {
      expect(wrapper.contains(<HeaderButtons />)).toBe(true);
    });
  });

  test('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
