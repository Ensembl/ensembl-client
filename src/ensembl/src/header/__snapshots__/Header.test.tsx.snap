// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<header>
  <div
    className="topBar"
  >
    <div>
      <div
        className="companyText"
      >
        Ensembl
      </div>
      <div
        className="strapline"
      >
        Pre-release - March 2019
      </div>
      <div
        className="copyright"
      >
        <a
          href="https://www.ebi.ac.uk"
        >
          Â© EMBL-EBI
        </a>
      </div>
    </div>
    <Connect(Component) />
  </div>
  <Connect(Component) />
  <Connect(Component) />
</header>
`;
